beacon-chain/powchain/block_reader.go

```diff
@@ -7,10 +7,20 @@ import (
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/shared/params"
 	"github.com/prysmaticlabs/prysm/shared/traceutil"
 	"go.opencensus.io/trace"
 )
 
+// searchThreshold to apply for when searching for blocks of a particular time. If the buffer
+// is exceeded we recalibrate the search again.
+const searchThreshold = 3
+
+// amount of times we repeat a failed search till is satisfies the conditional.
+const repeatedSearches = 2 * searchThreshold
```

Comment:
```suggestion
const repeatedSearches = 8 * searchThreshold
```
just to be sure we don't miss a block due to abnormality in block producing time

Reply:
I have let our searches be boundless, so we dont run into this.

Reply:
Context deadlines will be the fallback here, arbitary limits might allow some edge cases to slip through